python/TestHarness/testers/Tester.py

```diff
@@ -358,5 +359,10 @@ def checkRunnableBase(self, options, checks, test_list=None):
                 self.setStatus(reason, self.bucket_skip)
                 return False
 
+        # Check for display
+        if self.specs['display_required'] and os.environ['DISPLAY'] is None:
```

Comment:
Right, does this throw a key error if it's not there? Probably should make sure that's handled.